[게시글 생성/조회/수정/삭제] 슬로스 미션 제출합니다.

soojinroh opened the pull request.

indiveloper committed:
docs README.md 작성

indiveloper committed:
feat : 게시물의 정보를 담고 있는 bean 객체인 article class 구현

indiveloper committed:
docs: README.md 구현 기능 추가

soojinroh committed:
Update README.md

soojinroh committed:
Update README.md

soojinroh committed:
Update README.md

indiveloper committed:
feat 정적 파일 수정 시, 자동 재시작 및 LiveReload 설정

indiveloper committed:
feat 새로운 ArticleId를 생성하는 기능 구현

List<Article>에서 가장 큰 articleId보다 1만큼 큰 Long 타입 값을 반환

indiveloper committed:
fix: Article 클래스 수정

1. Article 클래스를 POJO로 수정
2. 불필요한 필드 삭제

indiveloper committed:
fix: articleId에 해당하는 Article 객체 조회 메서드 수정

indiveloper committed:
feat: 게시글 상세 조회 기능 구현

1. articleId에 해당하는 게시글 조회 메서드 생성
2. 게시글 상세 조회 웹 UI 수정

indiveloper committed:
feat: 게시글 상세 조회 기능 구현

1. articleId에 해당하는 게시글 상세 조회 메서드 구현
2. articleId에 해당하는 게시글을 가져와 Web UI에 구현

indiveloper committed:
feat: 웹 UI 수정

1. index.html, article.html, article-edit.html의 정적 파일 경로 수정
2. 전달받은 데이터를 타임리프를 이용하여 동적으로 출력하는 기능 적용

indiveloper committed:
feat : 게시물 수정 기능 구현

1. 수정하려는 게시글의 정보를 article-edit.html에 출력
2. 전달받은 Article 객체와 동일한 articleId를 가진 Article 객체를 List<Article>에서 바꾸는 메서드 구현
3. coverURL이 입력되지 않은 경우, 기본 이미지 URL 값을 할당하도록 수정

indiveloper committed:
refactor: 정적 파일의 경로를 수정

indiveloper committed:
refactor: 불필요한 파일 삭제

indiveloper committed:
feat : 환경설정 변경

indiveloper committed:
docs : README.md 수정

indiveloper committed:
feat : add lovely sloth image file

live like sloth, eat like sloth, sleep like sloth

indiveloper committed:
refactor : update REST 메서드 변경

1. post 방식의 메서드를 put으로 변경

indiveloper committed:
feat : 게시글 삭제 기능 구현

1. delete 메서드 방식을 이용해 게시글 삭제
2. article.html에 삭제 버튼 추가

indiveloper committed:
feat : 전체 게시글 목록 조회 기능 구현

1. get 메서드 방식을 이용해 전체 게시글 조회 메서드 구현
2. index.html에 전체 글 목록 출력

indiveloper committed:
refactor: css 오류 수정 및 게시글 상세 조회 페이지의 링크 경로 수정

indiveloper committed:
fix : fix conflict

indiveloper committed:
feat: ArticleRepository 테스트 추가

indiveloper committed:
refactor: ArticleController 클래스 테스트 리팩토링

1. 매직 넘버 상수화
2. response의 상태코드 확인 메서드 수정(is3xxRedirection() -> isFound())

indiveloper committed:
refactor : html 중복 제거

indiveloper committed:
refactor : convention 수정

1. 변수명 변경
2. 하드코딩한 값 상수화

indiveloper committed:
refactor : ArticleRepository의 DI를 필드 주입에서 생성자 주입으로 변경

pobiconan reviewed (changes requested):
spring 활용과 테스트 코드 구현 💯 
단, url path에 불필요한 부분들이 보이는데 url best pratice를 찾아 적용해 볼 것을 추천한다.

indiveloper committed:
refactor gradle dependencies 주입 방식 변경

gradle의 dependencies 주입 방식을 compile에서 implementation으로 변경

indiveloper committed:
refactor url path 변경

CRUD 기능을 드러내던 url을 REST API naming convention에 따라 CRUD 기능을 드러내지 않는 것으로 수정

indiveloper committed:
feat matchId 메서드 추가

매개변수로 주어진 articleId와 해당 article의 articleId가 일치하는지 확인하는 matchId 메서드 구현

indiveloper committed:
refactor 변수 타입 변경

Article 클래스의 articleId의 변수 타입을 Long에서 long으로 변경

indiveloper committed:
refactor Article class의 setter 삭제

indiveloper committed:
feat ArticleDto 구현

indiveloper committed:
refactor ModelAttribute를 Article에서 ArticleDto로 수정

indiveloper committed:
refactor updateArticle 메서드 수정

list의 set 메서드를 이용해 article의 내용을 수정하던 것에서 Article의 update 메서드를 이용해 article을 수정하는 것으로 변경

indiveloper committed:
fix article id 변경

articleId를 빈 문자열에서 0으로 변경

indiveloper committed:
refactor test 코드 수정

pobiconan reviewed (approved):
피드백 반영 잘 했네요. 💯 
수정과 관련해서 답변하나 남겼는데요. 추후 단계를 거듭하면서 이유에 대해 다시 한번 생각해 보면 좋겠네요.
